pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/download/SampleQueueTest.java

```diff
@@ -0,0 +1,178 @@
+            } else if (kitRequestShippingMap.get("ddpLabel").equals("errorKit")) {
+                assertEquals("GP manual Label", kitRequestShippingMap.get("sampleQueue"));
+            } else if (kitRequestShippingMap.get("ddpLabel").equals("newKit")) {
+                assertEquals("Waiting on GP", kitRequestShippingMap.get("sampleQueue"));
+            }else if (kitRequestShippingMap.get("ddpLabel").equals("shippedKit")) {
+                assertEquals("Shipped", kitRequestShippingMap.get("sampleQueue"));
+            }
+        });
+    }
+
+    private QueryParamsMap buildMockQueryParams(boolean onlyMostRecent, boolean humanReadable, String fileFormat) {
+        QueryParamsMap mainQueryParamsMap = mock(QueryParamsMap.class);
+
+        // Mocking the nested QueryParamsMap
+        QueryParamsMap nestedParentQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("parent")).thenReturn(nestedParentQueryParamsMap);
+        when(nestedParentQueryParamsMap.value()).thenReturn("participantList");
+
+        QueryParamsMap nestedOnlyMostRecentQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("onlyMostRecent")).thenReturn(nestedOnlyMostRecentQueryParamsMap);
+        when(nestedOnlyMostRecentQueryParamsMap.value()).thenReturn(String.valueOf(onlyMostRecent));
+
+        QueryParamsMap nestedRealmQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("realm")).thenReturn(nestedRealmQueryParamsMap);
+        when(nestedRealmQueryParamsMap.value()).thenReturn(ddpInstanceDto.getInstanceName());
+
+        QueryParamsMap nestedHumanReadableQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("humanReadable")).thenReturn(nestedHumanReadableQueryParamsMap);
+        when(nestedHumanReadableQueryParamsMap.value()).thenReturn(String.valueOf(humanReadable));
+
+        QueryParamsMap nestedUserIdQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("userId")).thenReturn(nestedUserIdQueryParamsMap);
+        when(nestedUserIdQueryParamsMap.value()).thenReturn(userId + "");
+
+        QueryParamsMap nestedFileFormatQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("fileFormat")).thenReturn(nestedFileFormatQueryParamsMap);
+        when(nestedFileFormatQueryParamsMap.value()).thenReturn(fileFormat);
+
+        QueryParamsMap nestedFiltersQueryParamsMap = mock(QueryParamsMap.class);
+        when(mainQueryParamsMap.get("filters")).thenReturn(nestedFileFormatQueryParamsMap);
+        when(nestedFileFormatQueryParamsMap.value()).thenReturn("[]");
+
+        return mainQueryParamsMap;
+    }
+
+    private ManualFilterParticipantList getFilterFromFile(String fileName){
+        try {
+            String filterJson = TestUtil.readFile(fileName);
+            ManualFilterParticipantList filterable = new Gson().fromJson(filterJson, ManualFilterParticipantList.class);
```

Comment:
Jackson did not work here, it could not understand the abstract class hierarchies in the filters. (This could also be me having very limited experience with Jackson)